### DISCUSSION TITLE
"Linting and checking validity of types ... Failed to compile. " - Unable to deploy

### DISCUSSION THREAD

**Discussion by gowthamc0702** [vercel/next.js#60770] [Help] [answered]
### Summary

Can anyone help me in this? i have stuck in this for a long time 

Github repo : https://github.com/gowthamc0702/vak

unable to deploy app , 
Type error: Page "app/products/[id]/page.tsx" does not match the required types of a Next.js Page.
"ProductPage" is not a valid Page export field.

![image](https://github.com/vercel/next.js/assets/108746137/d4988d10-a21a-4435-84e2-e76d32c97903)

**Comment by icyJoseph** [comment, marked as answer]
Hi,

You need to remove the `export const` for a `const` in:

- https://github.com/gowthamc0702/vak/blob/main/app/products/%5Bid%5D/page.tsx#L29

In App Router, page.tsx files can only export a few things, and `ProductPage` is not one of them.

**Reply by gowthamc0702** [reply]
I still have this error after that change , dont know why
![image](https://github.com/vercel/next.js/assets/108746137/c5dc4c73-70f3-41b8-b17c-ce340560cc19)

**Reply by icyJoseph** [reply]
Mmmm, I'd also remove the `use server`, that's not necessary there.

These errors should also be present locally, if you do `npm run build`, or yarn or whichever.

Do you still see it locally? 

Make sure you are looking at the right pipeline in the Vercel dashboard.

**Reply by gowthamc0702** [reply]
it runs fine on npm run dev , these errors are only in deployment

**Reply by icyJoseph** [reply]
No, not in `dev`, run `npm run build`, it is not the same thing, locally it should also fail. Again, `npm run build` not `dev`.

**Reply by gowthamc0702** [reply]
Generarting Static pages (3/9) [== ] , this ran continuosly and ended

![image](https://github.com/vercel/next.js/assets/108746137/af0d7e76-bc1d-4a06-ac2a-5ed0b73614ed)

**Reply by icyJoseph** [reply]
That means that it is crashing out and failing to generate the `/products` page. I see that you have a bug there:

```tsx
export default function ProductsList() {
  const list = Contents.Products;
  return (
    <>
      <h1>Products</h1>
      {list.length > 0 &&
        list.map(({ id, name, image }) => {
          <Product key={id} title={name} image={image} />; // you should return here!
        })}
    </>
  );
}
```

The fix:

```tsx
export default function ProductsList() {
  const list = Contents.Products;
  return (
    <>
      <h1>Products</h1>
      {list.length > 0 &&
        list.map(({ id, name, image }) => {
          return <Product key={id} title={name} image={image} />; // now we return
        })}
    </>
  );
}
```

**Reply by gowthamc0702** [reply]
i removed everything there except h1, still the same result
![image](https://github.com/vercel/next.js/assets/108746137/2bfd80ae-9da6-4dcc-9ccf-d851d5598236)

**Reply by icyJoseph** [reply]
Still no luck?

**Reply by gowthamc0702** [reply]
heyy @icyJoseph  , thanks a lot . the page is deployed (It had a static page issue )

**Comment by mannyamm21** [comment]
How you @gowthamc0702  solved the issue?
![image](https://github.com/user-attachments/assets/170eb9b2-75ac-40f1-8725-eae8a5066316)


this error is on npm run build

**Reply by VernSG** [reply]
Hello, have you solved this problem?
